### ISSUE TITLE
tooltip and popover flicker before correctly positioned

### ISSUE THREAD

**Issue by Rouche** [angular-ui/bootstrap#5678]
Hello,
After updating from 1.1.2 to 1.2.5 i have this same bug happening : https://github.com/angular-ui/bootstrap/issues/4407

It seems to happen when using popover-template that is not yet in template cache.
The plunker does not have the exact same issue. The arrow disapear and ficker a bit less.
The link is on the right end and better seen in a popup window preview.
http://plnkr.co/edit/Ej1NfYa3Oj75ZUv0vFVB?p=preview

Angular: 1.5.0
UIBS: 1.2.5
Bootstrap: 3.3.6

Edit: plunk changed for a cached template. so both examples are there.

**Comment by Foxandxss** [comment]
Will take a look, but the official angular team recommendation is to template cache all your templates before using it.

**Comment by Rouche** [comment]
Yes in production we will. .

**Comment by icfantv** [comment]
If you're doing it in production, you should also be doing it for development and test.  Short of initial file download size (or heavenforbid a dynamic template provider by your server), I can't think any good reason not to cache the templates in advance.  Modern build tools like webpack make doing this pretty painless.

**Comment by Rouche** [comment]
I agree. For dev its just annoying not having a builtin eclipse plugin and need to run a --watch and refresh, etc. But thats another story :)

**Comment by RobJacobs** [comment]
This has been discussed before and the conclusion was, if you don't have the popover templates loaded in the templateCache, this is the side effect.

**Comment by Rouche** [comment]
Ok i did not see this one. So its definatelly the reason.
Well i informed the QAs, so it should not be that bad now.

Closing.

**Comment by icfantv** [comment]
Hey @RobJacobs, I'm going to re-open this until we get a note in the docs about this.

**Comment by Rouche** [comment]
Thanks for answers!

**Comment by Jefftopia** [comment]
@RobJacobs @icfantv @Rouche  I don't think this is necessarily a templateCache issue, but I could be mistaken. Even after sending the template to cache I have a flickering tooltip on a right-floated button near the right-edge of the viewport.

I have verified that the template is chached by inspecting chromes's `chrome://cache` and found the template in the list. 

[Plunkr](http://plnkr.co/edit/eMulLM04mYtyr7yhAhl6?p=preview) to reproduce. 

Help would be great.

**Comment by Rouche** [comment]
The problem you have seems to be with the button css or something.

Check this one with a simple div:
http://plnkr.co/edit/ewp6NzFIOe1cOKgwcpje?p=preview

**Comment by Jefftopia** [comment]
@Rouche Since it doesn't seem to be related to your issue, I opened a new ticket [here](https://github.com/angular-ui/bootstrap/issues/5914).

In your example, the issue persists. Just remove the `float: right` from the button. When the div you added is at the right-edge, it also flickers as I described. 

Also, we can't let the css be the scapegoat. It's pure bootstrap 3.3.6. UI-Bootstrap has to work with UI CSS, otherwise, what's the point of the library?
